feature/DM-4962-opcua-gateway-java17 (#3812)
SoftwareAG/c8y-docs: content/change-logs/device-management/opcua-1022-0-0-opcua-gateway-java-17.md

---
date:
title: OPC UA gateway now runs on Java 17
product_area: Device management & connectivity
change_type:
  - value: change-2c7RdTdXo4
    label: Improvement
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-4962
version: 1022.0.0
---
Because Java 11 is outdated and no longer supported, the OPC UA gateway has been modernized and now runs on Java 17. 
This update enhances security and performance, ensuring the gateway remains reliable and efficient.

> Please ensure that your environment is compatible with Java 17 to take full advantage of the improvements. 

Reviewer: ```suggestion
Ensure that your environment is compatible with Java 17 to take full advantage of the improvements. 
```

> **Caution:** Please be aware that this update also replaces the internal gateway database and additional data migration 

Reviewer: ```suggestion

{{< c8y-admon-important >}}
Be aware that this update also replaces the internal gateway database and additional data migration 
```

> **Caution:** Please be aware that this update also replaces the internal gateway database and additional data migration 

Reviewer: See above.

or re-registration of device gateway may be required.

> Please refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) documentation section.

Reviewer: ```suggestion
Refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) in the documentation.
{{< /c8y-admon-important >}}

```

> Please refer to [Upgrading from 2021 to 2022 gateway version](/device-integration/opcua/#upgrade-2021-to-2022) documentation section.

Reviewer 2: ```suggestion
Please refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) documentation section.
```

---
date:
title: OPC UA gateway now runs on Java 17
product_area: Device management & connectivity
change_type:
  - value: change-2c7RdTdXo4
    label: Improvement
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-4962
version: 1022.0.0
---
Because Java 11 is outdated and no longer supported, the OPC UA gateway has been modernized and now runs on Java 17. 
This update enhances security and performance, ensuring the gateway remains reliable and efficient.
Ensure that your environment is compatible with Java 17 to take full advantage of the improvements. 

{{< c8y-admon-important >}}
Be aware that this update also replaces the internal gateway database and additional data migration 
or re-registration of device gateway may be required.

> Refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) in the documentation.

Reviewer: ```suggestion
```

> {{< /c8y-admon-important >}}

Reviewer: ```suggestion
{{< /c8y-admon-important >}}

Refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) in the documentation.
```
